lib/core/src/display/navigation/navigator.rs

```diff
@@ -0,0 +1,158 @@
+use crate::prelude::*;
+
+use super::EventHandler;
+use super::Event;
+use super::ZoomEvent;
+use super::PanEvent;
+use crate::system::web::EventListeningResult as Result;
+use crate::display::rendering::Camera;
+use crate::display::rendering::CameraType;
+use crate::display::rendering::DOMContainer;
+
+use nalgebra::Vector3;
+
+// =================
+// === Animation ===
+// =================
+
+/// Struct used to store animation related properties.
+struct Animation {
```

Comment:
This name is very misleading to what this struct does. Moreover, the doc are not descriptive, what are "animation related properties"? . Looking at the content of this struct, it containt physical properties for the object we animate, for the environment, and also animation state regarding where we want the object (camera) to travell. Maybe we should even refactor it to different structures, as it seems these information are very loosly coupled. Like, why we keep environment phgysical properties in the same struct as velocity of an object we animate?